tests/wpt/web-platform-tests/html/semantics/forms/the-input-element/type-change-state.html

```diff
@@ -49,12 +49,12 @@
           } else {
             input.value = "  foo\rbar  ";
             input.type = types[j].type;  // change state
-            if (types[i].type !== "color" && (types[j].sanitizedValue || types[j].sanitizedValue === "")) {
-              assert_equals(input.value, types[j].sanitizedValue, "input.value should be " + types[j].sanitizedValue + " after change of state");
+            if (types[j].sanitizedValue || types[j].sanitizedValue === "") {
```

Comment:
I haven't looked closely enough on this change yet; will need some time before I can determine whether this is the actual change we want.

Reply:
This is in #19330 
https://github.com/servo/servo/pull/19330/files#diff-a04db4d0de89c245a2878d38e99d51bdL52

will cause 
```
▶ Unexpected subtest result in /html/semantics/forms/the-input-element/type-change-state.html:
  │ FAIL [expected PASS] change state from color to date
  │   → assert_equals: input.value should be "#000000" after change of state expected "#000000" but got ""
  │ FAIL [expected PASS] change state from color to month
  │   → assert_equals: input.value should be "#000000" after change of state expected "#000000" but got ""
 ```

But actually it should pass

Reply:
You are correct. [This part of the spec](https://html.spec.whatwg.org/multipage/input.html#input-type-change) makes it clear that the value sanitization algorithm should be run when there is a change of state.

Reply:
Uh, sorry, that was not the full story. Because the value sanitization algorithm is ran when the `input.value` is set, color type inputs will sanitize their values, and if it's a `"  foo\rbar  "` string, it would instead set the value as `"#000000"`. This makes the expected value `"#000000"` even when you change the input type to text.

For date and month type inputs however, you're correct that it should expect the empty string instead.

I'll note that removing this check here isn't the correct fix though, we need to think of a more comprehensive way of testing this correctly -- the other PR is currently working on this.

Reply:
Since the other PR is already fixing this, we should revert the change here.